### PR TITLE
Awesome books with Javascript es6  features

### PR DESCRIPTION
**Pull request by Armynerh**
Setup my project:

Create a new directory and initialize it with git.
Use npm init -y command to create package.json file.
Create the entry point for  the JavaScript code called index.js
Create index.css file for the styles.
Create the main index.html file. Link the index.js and index.css files there.
Create a directory called modules.
Re-create the Awesome Books app following the rules:

Use ES6 syntax.
Divided code into modules.
Create a separate file for each module in the modules directory.
Import modules into the index.js file. 
Keep all functionalities of the app without errors.
Refactor the functions to arrow functions.
Use let and const in the correct way.
Display the current date below the navigation bar.
 Use the library [Luxon](https://moment.github.io/luxon/#/install?id=es6)
Download the library into the project using npm.
Import it as a module.
demonstrate how to use a new tool effectively with little externally curated support.

### CODE REVIEW

File: modules/dynamicDate.js [hunk 1]
```diff
@@ -0,0 +1,7 @@
+import { DateTime } from './luxon.min.js';
+
+export default function updateDateTime() {
+  const dateTimeElement = document.getElementById('date-para');
+  const newTime = DateTime.now();
+  dateTimeElement.textContent = newTime.toFormat('dd-MM-yyyy HH:mm:ss');
```
Comment (shahadat3669) [inline modules/dynamicDate.js]: - [x] 👍 Well! Functionally, you've successfully added the current date, and visually, the date is visible. However, ❌ your date and time format doesn't follow the wireframe. 😕 Please consider following the wireframe and updating your code to match the design as much as possible. 🔄

| Your | Design |
| --- | --- |
| image | image |

File: modules/manageBook.js [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+      removeButton.textContent = 'Remove';
+      removeButton.addEventListener('click', () => {
+        this.removeBook(index);
+      });
+      removeCell.appendChild(removeButton);
+      row.appendChild(removeCell);
+
+      bookTableBody.appendChild(row);
+    });
+  }
```
Comment (shahadat3669) [inline modules/manageBook.js:35]: - [x] 👍 Well! Your books are displaying perfectly when I click on the `List` nav item. However, there are three content sections in the wireframe: Books list, Add book form, and Contact info. Currently, you are showing only the title of the "books list" section by default. I recommend showing the table with the section title by default when there is at least one book, as per the project requirement. This way, users can see all the books instantly without having to click the "List" nav item. 📚📝

[Awesome-Books--JavaScript-Project.webm](https://github.com/Armynerh/awesome_books-es6/assets/55840999/28ad0414-3d0c-4df1-a39b-2de721303f0a)

File: index.html [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+               <div class="nav-items list-nav"><a href="#">List</a></div>
+               <div class="nav-items add-nav"><a href="#">Add new</a></div>
+               <div class="nav-items contact-nav"><a href="#">Contact</a></div>
+            </div>
+        </nav>
+       <div class="date"><p id="date-para"></p></div> 
+    </header>
+    <main>
+    <section>
+    <h1>Welcome to My bookstore</h1>
```
Comment (shahadat3669) [inline index.html:23]: - [x] 👍 Well! Upon reviewing your project, I noticed that the "Welcome to My Bookstore" h1 title is showing in all sections of the webpage. However, if you look closely at the design, this title should only be visible in the "Books list" section. Please make sure to follow the design carefully and update your code accordingly by updating all those section title. 📚🏷️

image

**Review by shahadat3669 — CHANGES_REQUESTED** [review]
Hi @Armynerh , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there! There are a few issues 🐛 that you still need to work 🔧 on though to go to the next project ⏭️ but you are almost there! 🏃



## Don't Worry 🙏

It's ok to have bugs in your code. 🆗 Please, see this code review as a chance for you to learn practical coding skills and learn new tips that can improve your code quality 💯 and help you **get a better ** 💼.

I am going to guide you on how to fix them down below. 👇 


## Highlights
✅ No linter Errors
✅ Descriptive PR Title
✅ Descriptive PR Description
✅ The code is top-notch! 💯


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

## Have any doubt ❓

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.


## Thank you for your time 🙏 and call me if you need help or clarification ☎️
| GitHub | LinkedIn |  Slack |  E-Mail    |
| --- | --- | --- | --- |
|   [ GitHub ](https://github.com/shahadat3669)   |   [ LinkedIn ](https://www.linkedin.com/in/shahadat-cseng/)    |  [ Slack ](https://app.slack.com/client/T47CT8XPG/D05GT4E4WAY/rimeto_profile/U04426SD71V)   |  [ Gmail ](mailto:shahadat3adsf669@gmail.com?subject=Microverse%20Code%20Review) |

## ⚠️ WARNING ⚠️
**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by frempongdev — APPROVED** [review]
# STATUS: APPROVED 🟢 ✔️  

Hi @Armynerh, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights
linter checks :heavy_check_mark: 
all application features work :heavy_check_mark:
all corrections made :heavy_check_mark: 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**